packages/cli/src/utils/election.ts

```diff
@@ -8,22 +11,35 @@ import { bitIsSet, parseBlockExtraData } from '@celo/utils/lib/istanbul'
 export class ElectionResultsCache {
   private readonly cache = new Map()
 
-  constructor(private readonly election: any, private readonly epochSize: number) {}
+  constructor(
+    private readonly kit: ContractKit,
+    private readonly isCel2: boolean,
+    private readonly electionWrapper: ElectionWrapper,
+    private readonly epochManagerWrapper?: EpochManagerWrapper
+  ) {
+    if (isCel2 && !epochManagerWrapper) {
+      throw new Error('EpochManagerWrapper is required for L2')
+    }
+  }
 
   /**
    * Returns the list of elected signers for a given block.
    * @param blockNumber The block number to get elected signers for.
    */
   async electedSigners(blockNumber: number): Promise<Address[]> {
-    const epoch = this.epochNumber(blockNumber)
+    const epoch = await this.kit.getEpochNumberOfBlock(blockNumber)
     const cached = this.cache.get(epoch)
     if (cached) {
       return cached
     }
-    // For the first epoch, the contract might be unavailable
-    const electedSigners = await this.election.getValidatorSigners(
-      epoch === 1 ? blockNumber : this.firstBlockOfEpoch(epoch)
-    )
+
+    const electedSigners = this.isCel2
+      ? await this.epochManagerWrapper!.getElectedSigners()
+      : await this.electionWrapper.getValidatorSigners(
+          // For the first epoch, the contract might be unavailable
+          epoch === 1 ? blockNumber : await this.kit.getFirstBlockNumberForEpoch(epoch)
```

Comment:
you know we can just remove this if epoch is 1 stuff because its not april 2020 anymore ;)

Reply:
ok but do we still have historically data? i thought you said support was removed?